### PR TITLE
Add RocketMQ 5.x metrics definition.

### PR DESCRIPTION
**Pull request by yangwenting-ywt**
Please do not create a Pull Request without creating an issue first. 

## What is the purpose of the change

Add documentation for RocketMQ 5.0 version metrics definitions and related scenarios.

### CODE REVIEW

**Review by zhouxinyu — APPROVED** [review]
Looks great, thanks for your contributions.

**Comment by chenzlalvin** [comment]
> Please do not create a Pull Request without creating an issue first.
> 
> ## What is the purpose of the change
> Add documentation for RocketMQ 5.0 version metrics definitions and related scenarios.

A tip. please add comments about the support versions for this metrics definition.

**Comment by aaron-ai** [comment]
The improvement of observability is a very crucial new feature of RocketMQ 5.0 compared to 4.0. Sincerely appreciate your contribution towards it.

**Comment by ShadowySpirits** [comment]
> > Please do not create a Pull Request without creating an issue first.
> > ## What is the purpose of the change
> > Add documentation for RocketMQ 5.0 version metrics definitions and related scenarios.
> 
> A tip. please add comments about the support versions for this metrics definition.

Agree, metrics for RocketMQ were introduced since 5.1.0 and only support the broker.

File: i18n/en/docusaurus-plugin-content-docs/version-5.0/12-observability/01metrics.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+### Metric types
+The standard for defining metrics in RocketMQ complies with that for defining the metrics in open source Prometheus. The metric types that RocketMQ offers include counters, gauges, and histograms. For more information, see [METRIC TYPES](https://prometheus.io/docs/concepts/metric_types/).
+
+### Broker metrics
+The following table describes the labels of the metrics that are related to the Message Queue for Apache RocketMQ broker.
+
+- cluster: RocketMQ cluster name.
+- node_type: the type of service node, whitch includes the following:proxy,broker,nameserver.
+- node_id:the ID of the service node.
```
Comment (ShadowySpirits) [inline i18n/en/docusaurus-plugin-content-docs/version-5.0/12-observability/01metrics.md]: There should be a space between `node_id:` and `the ID`.

File: i18n/en/docusaurus-plugin-content-docs/version-5.0/12-observability/01metrics.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+| gauge | rocketmq_thread_pool_wartermark | count | The number of tasks queued in the thread pool. | cluster,node_type,node_id,name |
+
+
+
+### Producer metrics
+The following table describes the labels of the metrics that are related to the producers in Message Queue for Apache RocketMQ.
+
+- cluster: RocketMQ cluster name.
+- node_type: the type of service node, whitch includes the following:proxy,broker,nameserver.
+- node_id:the ID of the service node.
```
Comment (ShadowySpirits) [inline i18n/en/docusaurus-plugin-content-docs/version-5.0/12-observability/01metrics.md]: Ditto.

File: versioned_docs/version-5.0/12-observability/01metrics.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# Metrics
+RocketMQ以Prometheus格式公开以下指标。您可以使用这些指标监视您的集群。
```
Comment (ShadowySpirits) [inline versioned_docs/version-5.0/12-observability/01metrics.md]: It's better that add space between Chinese words and English words.

File: i18n/en/docusaurus-plugin-content-docs/version-5.0/12-observability/01metrics.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+- topic: the topic of Message Queue for Apache RocketMQ.
+- message_type: the type of a message, which includes the following:<br />Normal:normal messages;<br />FIFO:ordered messages;<br />Transaction:Transactional messages;<br />Delay:scheduled or delayed messages.
```
Comment (ShadowySpirits) [inline i18n/en/docusaurus-plugin-content-docs/version-5.0/12-observability/01metrics.md]: I think the value of label message_type is lowercase.

https://github.com/apache/rocketmq/blob/c9975961f81292f62c2c47f9e58f64472653ea7e/common/src/main/java/org/apache/rocketmq/common/attribute/TopicMessageType.java#L60-L62

File: versioned_docs/version-5.0/12-observability/01metrics.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+| counter | rocketmq_send_to_dlq_messages_total | count | 转为死信状态的消息量。（变成死信状态指的是消息达到最大重投次数后不再投递） | cluster,node_type,node_id,topic, consumer_group |
+| histogram | rocketmq_rpc_latency | millisecond | rpc 调用耗时：<br />le_1_ms<br />le_3_ms<br />le_5_ms<br />le_10_ms<br />le_100_ms<br />le_1_s<br />le_3_s<br />le_overflow | cluster,node_typ,node_id,protocol_type,request_code,response_code |
+| gauge | rocketmq_storage_size | byte | 存储大小，包含所有文件。| cluster,node_type,node_id |
+| counter | rocketmq_storage_read_bytes_total | byte | 储存层读取数据量。 | cluster,node_type,node_id,topic |
+| gauge | rocketmq_storage_read_bytes_max | byte | 储存层每秒读取数据量峰值。 | cluster,node_type,node_id,topic |
+| counter | rocketmq_storage_write_bytes_total | byte | 储存层写入数据量。| cluster,node_type,node_id,topic |
+| gauge | rocketmq_storage_write_bytes_max | byte | 储存层每秒写入数据量峰值。  | cluster,node_type,node_id,topic |
+| Histogram | rocketmq_storage_write_latency | millisecond | 储存层写入消息耗时：<br />le_1_ns: ≤ 1 ns<br /> le_10_ns: ≤ 10 ns<br />le_100_ns: ≤ 100 ns<br />le_1000_ns: ≤ 1 ms<br />le_overflow: > 1 ms | cluster,node_type,node_id,topic,message_type |
```
Comment (ShadowySpirits) [inline versioned_docs/version-5.0/12-observability/01metrics.md]: We haven't implemented these metrics for now.

File: versioned_docs/version-5.0/12-observability/01metrics.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+### 消息堆积场景
+![RocketMQ 队列消息状态](../picture/v5/observability/message-accumulation.png)<br />上图表示指定主题的某一队列中各消息的状态，消息队列RocketMQ版将处于不同处理阶段的消息数量和耗时进行统计，这些指标可直接反映队列中消息的处理速率和堆积情况，通过观察这些指标可初步判断业务的消费是否异常。具体的指标含义和计算公式如下：
+
+| 名称 | 描述 | 计算公式 |
+| --- | --- | --- |
+| 处理中的消息（inflight messages） | 在消费者客户端正在处理，但客户端还未返回消费结果的消息。 | 最晚一条拉取消息的位点－最晚一条提交消息的位点 |
+| 已就绪消息（ready messages） | 消息在消息队列RocketMQ版服务端已就绪，对消费者可见可被消费的消息。 | 最大消费位点－最晚一条拉取消息的位点 |
+| 已就绪消息的就绪时间（ready time） | 普通消息、顺序消息：消息存储到服务端的时间。 <br /> 定时/延时消息：消息定时或延时结束的时间。 <br />  事务消息：事务提交的时间。 | -- |
+| 已就绪消息的排队时间（ready message queue time）| 该时间反应消费者拉取消息的及时性。 | 当前时间－最早一条就绪消息的就绪时间 |
+| 消费处理滞后时间（comsumer lag time） | 该时间反应消费者完成消息处理的及时性。 | 当前时间－最早一条未提交消息的就绪时间|
```
Comment (ShadowySpirits) [inline versioned_docs/version-5.0/12-observability/01metrics.md]: comsumer -> consumer

**Review by ShadowySpirits — CHANGES_REQUESTED** [review]

**Review by aaron-ai — APPROVED** [review]

**Pull request merged**